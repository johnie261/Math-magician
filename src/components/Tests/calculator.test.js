import '@testing-library/jest-dom';
import * as React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import Calculator from '../calculator';

describe('calculator component', () => {
  test('renders calculator buttons', () => {
    render(<Calculator />);

    const zeros = screen.getAllByText('0');
    expect(zeros[1]).toBeInTheDocument();
    expect(screen.getByText('1')).toBeInTheDocument();
    expect(screen.getByText('2')).toBeInTheDocument();
    expect(screen.getByText('3')).toBeInTheDocument();
    expect(screen.getByText('4')).toBeInTheDocument();
    expect(screen.getByText('5')).toBeInTheDocument();
    expect(screen.getByText('6')).toBeInTheDocument();
    expect(screen.getByText('7')).toBeInTheDocument();
    expect(screen.getByText('8')).toBeInTheDocument();
    expect(screen.getByText('9')).toBeInTheDocument();
    expect(screen.getByText('AC')).toBeInTheDocument();
    expect(screen.getByText('+/-')).toBeInTheDocument();
    expect(screen.getByText('%')).toBeInTheDocument();
    expect(screen.getByText('รท')).toBeInTheDocument();
    expect(screen.getByText('-')).toBeInTheDocument();
    expect(screen.getByText('+')).toBeInTheDocument();
    expect(screen.getByText('.')).toBeInTheDocument();
    expect(screen.getByText('=')).toBeInTheDocument();
  });
  test('it updates the display when the button is clicked', () => {
    render(<Calculator />);

    fireEvent.click(screen.getByText('5'));
    fireEvent.click(screen.getByText('+'));
    fireEvent.click(screen.getByText('3'));
    fireEvent.click(screen.getByText('='));

    expect(document.querySelector('.calc-output').textContent).toBe('8');
  });
});
