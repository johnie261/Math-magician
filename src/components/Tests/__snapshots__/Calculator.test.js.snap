// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="Container"
>
  <div
    className="calc-container"
  >
    <div
      className="calc-output"
    >
      0
    </div>
    <div
      className="calc-keys"
    >
      <div
        className="first-row"
      >
        <input
          className="btn"
          name=""
          onClick={[Function]}
          type="button"
          value="AC"
        />
        <input
          className="btn"
          name=""
          onClick={[Function]}
          type="button"
          value="+/-"
        />
        <input
          className="btn"
          name=""
          onClick={[Function]}
          type="button"
          value="%"
        />
        <input
          className="btn color"
          name=""
          onClick={[Function]}
          type="button"
          value="รท"
        />
      </div>
      <div
        className="second-row"
      >
        <input
          className="btn"
          name=""
          onClick={[Function]}
          type="button"
          value="7"
        />
        <input
          className="btn"
          name=""
          onClick={[Function]}
          type="button"
          value="8"
        />
        <input
          className="btn"
          name=""
          onClick={[Function]}
          type="button"
          value="9"
        />
        <input
          className="btn color"
          name=""
          onClick={[Function]}
          type="button"
          value="x"
        />
      </div>
      <div
        className="third-row"
      >
        <input
          className="btn"
          name=""
          onClick={[Function]}
          type="button"
          value="4"
        />
        <input
          className="btn"
          name=""
          onClick={[Function]}
          type="button"
          value="5"
        />
        <input
          className="btn"
          name=""
          onClick={[Function]}
          type="button"
          value="6"
        />
        <input
          className="btn color"
          name=""
          onClick={[Function]}
          type="button"
          value="-"
        />
      </div>
      <div
        className="fourth-row"
      >
        <input
          className="btn"
          name=""
          onClick={[Function]}
          type="button"
          value="1"
        />
        <input
          className="btn"
          name=""
          onClick={[Function]}
          type="button"
          value="2"
        />
        <input
          className="btn"
          name=""
          onClick={[Function]}
          type="button"
          value="3"
        />
        <input
          className="btn color"
          name=""
          onClick={[Function]}
          type="button"
          value="+"
        />
      </div>
      <div
        className="fifth-row"
      >
        <input
          className="zeroBtn"
          name=""
          onClick={[Function]}
          type="button"
          value="0"
        />
        <input
          className="btn"
          name=""
          onClick={[Function]}
          type="button"
          value="."
        />
        <input
          className="btn color"
          name=""
          onClick={[Function]}
          type="button"
          value="="
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`matches snapshots 1`] = `
<div
  className="Container"
>
  <div
    className="calc-container"
  >
    <div
      className="calc-output"
    >
      0
    </div>
    <div
      className="calc-keys"
    >
      <div
        className="first-row"
      >
        <input
          className="btn"
          name=""
          onClick={[Function]}
          type="button"
          value="AC"
        />
        <input
          className="btn"
          name=""
          onClick={[Function]}
          type="button"
          value="+/-"
        />
        <input
          className="btn"
          name=""
          onClick={[Function]}
          type="button"
          value="%"
        />
        <input
          className="btn color"
          name=""
          onClick={[Function]}
          type="button"
          value="รท"
        />
      </div>
      <div
        className="second-row"
      >
        <input
          className="btn"
          name=""
          onClick={[Function]}
          type="button"
          value="7"
        />
        <input
          className="btn"
          name=""
          onClick={[Function]}
          type="button"
          value="8"
        />
        <input
          className="btn"
          name=""
          onClick={[Function]}
          type="button"
          value="9"
        />
        <input
          className="btn color"
          name=""
          onClick={[Function]}
          type="button"
          value="x"
        />
      </div>
      <div
        className="third-row"
      >
        <input
          className="btn"
          name=""
          onClick={[Function]}
          type="button"
          value="4"
        />
        <input
          className="btn"
          name=""
          onClick={[Function]}
          type="button"
          value="5"
        />
        <input
          className="btn"
          name=""
          onClick={[Function]}
          type="button"
          value="6"
        />
        <input
          className="btn color"
          name=""
          onClick={[Function]}
          type="button"
          value="-"
        />
      </div>
      <div
        className="fourth-row"
      >
        <input
          className="btn"
          name=""
          onClick={[Function]}
          type="button"
          value="1"
        />
        <input
          className="btn"
          name=""
          onClick={[Function]}
          type="button"
          value="2"
        />
        <input
          className="btn"
          name=""
          onClick={[Function]}
          type="button"
          value="3"
        />
        <input
          className="btn color"
          name=""
          onClick={[Function]}
          type="button"
          value="+"
        />
      </div>
      <div
        className="fifth-row"
      >
        <input
          className="zeroBtn"
          name=""
          onClick={[Function]}
          type="button"
          value="0"
        />
        <input
          className="btn"
          name=""
          onClick={[Function]}
          type="button"
          value="."
        />
        <input
          className="btn color"
          name=""
          onClick={[Function]}
          type="button"
          value="="
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`renders correctly 1`] = `
<div
  style={
    Object {
      "marginTop": "10rem",
      "textAlign": "center",
    }
  }
>
   
  <h1>
    Quote of the day
  </h1>
  <h2
    className="load"
  >
    Loading...
  </h2>
</div>
`;
